x-pack/plugins/ingest_manager/server/services/agents/enroll.ts

```diff
@@ -62,9 +62,14 @@ export async function enroll(
 
   const accessAPIKey = await APIKeyService.generateAccessApiKey(soClient, agent.id, configId);
 
-  await soClient.update<AgentSOAttributes>(AGENT_SAVED_OBJECT_TYPE, agent.id, {
-    access_api_key_id: accessAPIKey.id,
-  });
+  await soClient.update<AgentSOAttributes>(
+    AGENT_SAVED_OBJECT_TYPE,
+    agent.id,
+    {
+      access_api_key_id: accessAPIKey.id,
+    },
+    { refresh: false }
```

Comment:
You probably need to update the checkin code to make the tests green, currently we are making a search using the  `access_api_key_id` during the checkin so if the agent is not refreshed the search will not work.

We can change that in the checkin to get the agent by id (the id is in the request params) and check if the access_api_key_id match.

Reply:
Thanks, @nchaulet that makes sense. I'll look into it. I'm also curious/concerned why these pass locally but fail in CI

Reply:
My guess is your macbook is faster than the CI server and the refresh is done before the checkin call.